pygeo/parameterization/DVGeoSketch.py

```diff
@@ -0,0 +1,281 @@
+"""
+BaseDVGeo
+
+Holds a basic version of a DVGeo geometry object to be used with a parametric geometry engine
+Enables the use of ESP (Engineering Sketch Pad) and OpenVSP (Open Vehicle Sketch Pad) with the MACH-Aero framework
+
+"""
+
+# ======================================================================
+#         Imports
+# ======================================================================
+from abc import abstractmethod
+from collections import OrderedDict
+from mpi4py import MPI
+from .BaseDVGeo import BaseDVGeometry
+from pyspline.utils import openTecplot, closeTecplot, writeTecplot1D
+
+
+class DVGeoSketch(BaseDVGeometry):
+    """A class for manipulating parametric geometry
+
+    The purpose of the BaseDVGeoSketchy class is to provide translation
+    of the ESP/OpenVSP geometry engine to externally supplied surfaces. This
+    allows the use of ESP/OpenVSP design variables to control the MACH
+    framework.
+
+    There are several import limitations:
+
+    1. Since ESP and OpenVSP are surface based only, they cannot be used to
+    parameterize a geometry that doesn't lie on the surface. This
+    means it cannot be used for structural analysis. It is generally
+    possible use most of the constraints DVConstraints since most of
+    those points lie on the surface.
+
+    2. It cannot handle *moving* intersection. A geometry with static
+    intersections is fine as long as the intersection doesn't move
+
+    3. It does not support complex numbers for the complex-step method.
+
+    4. It does not support separate configurations.
+
+    5. Because of limitations with ESP and OpenVSP, this class
+    uses parallel finite differencing to obtain the required Jacobian
+    matrices.
+
+    Parameters
+    ----------
+    fileName : str
+       filename of .vsp3 file (OpenVSP) or .csm file (ESP).
+
+    comm : MPI Intra Comm
+       Comm on which to build operate the object. This is used to
+       perform embarrassingly parallel finite differencing. Defaults to
+       MPI.COMM_WORLD.
+
+    scale : float
+       A global scale factor from the ESP/VSP geometry to incoming (CFD) mesh
+       geometry. For example, if the ESP/VSP model is in inches, and the CFD
+       in meters, scale=0.0254.
+
+    """
+
+    def __init__(self, fileName, comm=MPI.COMM_WORLD, scale=1.0, projTol=0.01):
+        super().__init__(fileName=fileName)
+
+        # this scales coordinates from model to mesh geometry
+        self.modelScale = scale
+        # and this scales coordinates from mesh to model geometry
+        self.meshScale = 1.0 / scale
+        self.projTol = projTol * self.meshScale  # default input is in meters.
+
+        self.updatedJac = {}
+        self.comm = comm
+
+        # Initial list of DVs
+        self.DVs = OrderedDict()
+
+    @abstractmethod
+    def addPointSet(self, points, ptName, **kwargs):
```

Comment:
Hmm, is there a way to inherent from the baseclass without redefining them here. 
I thought abstract methods only cause errors if you try to instantiate an object, so we should be fine to remove these abstract methods from this base class since it should never be used directly right?

Reply:
```python
from abc import abstractmethod
from abc import ABC
 
class A(ABC):
    
    @abstractmethod
    def test(self):
        print('this shouldnt run...')
        pass 
    
class B(A):
    pass

class C(B):
    
    def test(self):
        print('hi')
        
        
myC = C()
myC.test()

myB = B()
myA = A()
```
I made a little test script to show you what I mean. 
It should fail at B() even though B has no virtual methods itself.

Reply:
In this case we can remove this I think, because
1. B is also an ABC so it will not be instantiated
2. A already defines `addPointSet` and this implementation does not add anything

Reply:
Duplicated abstract methods have been removed from the intermediate class